### DISCUSSION TITLE
installing duckdb is broken

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#5210] [General]
it seems binary file is not available

Requirement already satisfied: duckdb in c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages (0.5.2.dev1614)
Collecting duckdb
  Using cached duckdb-0.5.2.dev1616.tar.gz (9.9 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy>=1.14 in c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages (from duckdb) (1.23.1)
Building wheels for collected packages: duckdb
  Building wheel for duckdb (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\__init__.py -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      running egg_info
      writing duckdb.egg-info\PKG-INFO
      writing dependency_links to duckdb.egg-info\dependency_links.txt
      writing requirements to duckdb.egg-info\requires.txt
      writing top-level names to duckdb.egg-info\top_level.txt
      reading manifest file 'duckdb.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.h' under directory 'src'
      writing manifest file 'duckdb.egg-info\SOURCES.txt'
      copying duckdb_query_graph\parse_profiling_output.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\query_graph.css -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\raphael.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\treant.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      creating build\lib.win-amd64-cpython-310\duckdb-stubs
      copying duckdb-stubs\__init__.pyi -> build\lib.win-amd64-cpython-310\duckdb-stubs
      running build_ext
      building 'duckdb' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for duckdb
  Running setup.py clean for duckdb
Failed to build duckdb
Installing collected packages: duckdb
  Attempting uninstall: duckdb
    Found existing installation: duckdb 0.5.2.dev1614
    Uninstalling duckdb-0.5.2.dev1614:
      Successfully uninstalled duckdb-0.5.2.dev1614
  Running setup.py install for duckdb ... error
  error: subprocess-exited-with-error

  × Running setup.py install for duckdb did not run successfully.
  │ exit code: 1
  ╰─> [29 lines of output]
      C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running install
      C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\__init__.py -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      running egg_info
      writing duckdb.egg-info\PKG-INFO
      writing dependency_links to duckdb.egg-info\dependency_links.txt
      writing requirements to duckdb.egg-info\requires.txt
      writing top-level names to duckdb.egg-info\top_level.txt
      reading manifest file 'duckdb.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.h' under directory 'src'
      writing manifest file 'duckdb.egg-info\SOURCES.txt'
      copying duckdb_query_graph\parse_profiling_output.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\query_graph.css -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\raphael.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      copying duckdb_query_graph\treant.js -> build\lib.win-amd64-cpython-310\duckdb_query_graph
      creating build\lib.win-amd64-cpython-310\duckdb-stubs
      copying duckdb-stubs\__init__.pyi -> build\lib.win-amd64-cpython-310\duckdb-stubs
      running build_ext
      building 'duckdb' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  Rolling back uninstall of duckdb
  Moving to c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages\duckdb-0.5.2.dev1614.dist-info\
   from C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\Lib\site-packages\~uckdb-0.5.2.dev1614.dist-info
  Moving to c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages\duckdb-stubs\
   from C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\Lib\site-packages\~uckdb-stubs
  Moving to c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages\duckdb.cp310-win_amd64.pyd
   from C:\Users\mimoune.djouallah\AppData\Local\Temp\pip-uninstall-rc9bvzg4\duckdb.cp310-win_amd64.pyd
  Moving to c:\users\mimoune.djouallah\appdata\local\programs\python\python310\lib\site-packages\duckdb_query_graph\
   from C:\Users\mimoune.djouallah\AppData\Local\Programs\Python\Python310\Lib\site-packages\~4ckdb_query_graph
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> duckdb

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: python.exe -m pip install --upgrade pip

**Comment by Mause** [comment]
As the DuckDB builds take several hours to fully complete, we're not able to upload all of the prebuilt binaries simultaneously. If you're having issues, I would recommend installing the second or third newest version, or passing the `--only-binary :all:` option to pip 

See https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-only-binary for more details
